lms/djangoapps/completion/services.py

```diff
@@ -54,8 +54,29 @@ def get_completions(self, candidates):
             course_key=self._course_key,
             block_key__in=candidates,
         )
-        completions = {block.block_key: block.completion for block in completion_queryset}
+        completions = {block.full_block_key: block.completion for block in completion_queryset}
         for candidate in candidates:
             if candidate not in completions:
                 completions[candidate] = 0.0
         return completions
+
+    def get_completion_for_vertical(self, item):
+        """
+        Calculates and returns whether a particular vertical is complete.
+        The logic in this method is temporary, and will go away once the
+        completion API is able to store a first-order notion of completeness
+        for parent blocks (right now it just stores completion for leaves-
+        problems, HTML, video, etc.).
+        """
+        complete = None
```

Comment:
Do we need validation logic here to ensure the passed in `item` is a block of `vertical` type?